### DISCUSSION TITLE
Include custom <script> in <head> section for GDPR conformity

### DISCUSSION THREAD

**Discussion by mgloec** [mastodon/mastodon#19794] [General]
I am not very familiar with react development so maybe my description is a little bit unusual (sorry):
To be compliant with GDPR I want to include some custom script tags in the main <head of mastodon. Is there a way to do it? In my case it would be:

First script in head> tag:
```
<script id="Cookiebot" src="https://consent.cookiebot.com/uc.js" data-cbid="xxxxxxxxxxxxxxxxxxxxx" data-blockingmode="auto" type="text/javascript"></script>
```
And maybe in the main footer section of mastodon:

```
<script id="CookieDeclaration" src="https://consent.cookiebot.com/xxxxxxxxxxxxx/cd.js" type="text/javascript" async></script>
```

Is there any way to include this in the source code for me, in my current installation?
Is there maybe somthing similiar planned as feature?

Thank you!

**Comment by LettError** [comment]
I'm interested in this as well. How does the Mastodon community take care of this? Any recommendations or best practice?

**Comment by Gargron** [comment]
You do not need a cookie banner on Mastodon because Mastodon does not include any 3rd party or tracking code. Mastodon cookies only serve the essential functionality of the web interface.

**Reply by DarkShyMW** [reply]
but if i need to use 3-rd party codes, what i should to do?

**Comment by LettError** [comment]
(Ha! Thank you @gargron for all your work on this!)
I have some reports from a very concerned user, and a quick (admittedly free) audit from cookiebot.com says our site is not compliant. I see the Mastodon instances operated (EUIPO ad EDPS) by the EU do not have a cookie banners either, but I'm open for guidance.
